py/gc.c

```diff
@@ -820,10 +839,16 @@ void gc_dump_alloc_table(void) {
                 while (bl2 < MP_STATE_MEM(gc_alloc_table_byte_len) * BLOCKS_PER_ATB && ATB_GET_KIND(bl2) == AT_FREE) {
                     bl2++;
                 }
-                if (bl2 - bl >= 2 * DUMP_BYTES_PER_LINE) {
+                size_t skip = bl2 - bl;
+                size_t lines = skip / DUMP_BYTES_PER_LINE;
+                if (lines >= 2) {
                     // there are at least 2 lines containing only free blocks, so abbreviate their printing
-                    mp_printf(&mp_plat_print, "\n       (%u lines all free)", (uint)(bl2 - bl) / DUMP_BYTES_PER_LINE);
-                    bl = bl2 & (~(DUMP_BYTES_PER_LINE - 1));
+                    mp_printf(&mp_plat_print, "\n       (%u lines all free)", (uint)lines);
+                    skip = lines * DUMP_BYTES_PER_LINE;
+                    #if DUMP_FRAGMENTATION_INFO
```

Comment:
This doesn't get reached if DUMP_BLOCK_INFO is 0, is this the intent?

Reply:
Yeah that is the intent. This line is accounting for the fact that skipping lines means we wouldn't otherwise be counting these free blocks in the current run of free blocks.

Reply:
Sorry I didn't look at the code in detail so I 'm not 100% sure I understand what this means :)
But I'll rephrase my question: when DUMP_FRAGMENTATION_INFO is 1 and DUMP_BLOCK_INFO is 0, `n` seems to go through different caluclations than in the case where both DUMP_FRAGMENTATION_INFO and DUMP_BLOCK_INFO are 1. But the end result is the same in both cases, right?